go.mod

```
@@ -1,6 +1,6 @@
 module github.com/coder/terraform-provider-coder/v2
 
-go 1.22.9
+go 1.24.1
```

```suggestion
go 1.24.2
```

Any objection to jumping straight to the latest? It's a security release.
https://go.dev/doc/devel/release#go1.24.minor